### DISCUSSION TITLE
Updating user from Node and Local Storage

### DISCUSSION THREAD

**Discussion by irakli2206** [pocketbase/pocketbase#1273] [Q&A] [answered]
I made a Stripe webhook in NextJS Node environment, where after a successful subscription payment I want to update the PB user and set subscriptionStatus field to Active. This part is clear, however, after updating from Node, the Local Storage of course stays the same as it was before and thus the UI does not get updated. What is the best approach to update the Local Storage user after the user gets updated in PocketBase?

**Comment by ganigeorgiev** [comment, marked as answer]
There are several options:

1. Call `await pb.collection('users').authRefresh()` on every full page load in the browser or after the Stripe webhook redirect.

2. Listen to realtime changes of the user data:
  ```js
  pb.collection('users').subscribe(pb.authStore.model.id, (e) => {
    console.log(e.action);
    console.log(e.record);

    // refresh the authstore with the latest record changes
    // eventually you may also call authRefresh() allowing you to fetch your user with some expanded relations
    pb.authStore.save(pb.authStore.token, e.record);
  });
  ```

**Reply by irakli2206** [reply]
Which would you say is the recommended way to do it?

**Reply by ganigeorgiev** [reply]
It depends on your use case.

Personally I would choose the first option (aka. calling `authRefresh()`) since it is more "lightweight" and easier to understand.
